lib/chef/client.rb

```diff
@@ -515,6 +517,48 @@ def setup_run_context
     end
 
     #
+    # Adds a required recipe as specified by the Chef Server
+    #
+    # @return The modified run context
+    #
+    # @api private
+    #
+    # TODO: @rest doesn't appear to be used anywhere outside
+    # of client.register except for here. If it's common practice
+    # to create your own rest client, perhaps we should do that
+    # here but it seems more appropriate to reuse one that we
+    # know is already created. for ease of testing, we'll pass
+    # the existing rest client in as a parameter
```

Comment:
The comments around `#rest_clean` indicate that it's used for API reporters. If it is indeed related to JSON sanitization, that's also non-applicable here because neither the request nor the response contain any JSON. This is simply a `GET` to retrieve a recipe file.